### DISCUSSION TITLE
How to deploy a trained model?

### DISCUSSION THREAD

**Discussion by catastropiyush** [mir-group/nequip#445] [Q&A]
Hello,
I just started out with using NequIP.
`$ nequip-train configs/example.yaml` After running this I get the following files in the directory.
![343614149-b59b19e6-e965-4f48-ad3b-bb2320b7223a](https://github.com/user-attachments/assets/01bd565e-06a8-4e45-b7a8-c8a07e426a61)

But how do I deploy the trained model for LAMMPS?
`nequip-deploy build path/to/training/session/ path/to/deployed.pth`
What should one put in `path/to/training/session` the directory that contains the` .yaml` file?

```
pair_style	nequip
pair_coeff	* * deployed.pth <NequIP type for LAMMPS type 1> <NequIP type for LAMMPS type 2> ...
```

What are LAMMPS type 1 and LAMMPS type 2 here?

I apologize if the questions are too basic.
Thank you for your time

**Comment by Linux-cpp-lisp** [comment]
Hi @catastropiyush ,

Thanks for your interest in `nequip`! The Allegro tutorial (https://github.com/mir-group/allegro?tab=readme-ov-file#tutorial) should address these questions.

**Reply by catastropiyush** [reply]
Thank you for replying, I was running the  colab  on a T4 GPU and I am getting the following error.
After running the evaluate test error command
`!nequip-evaluate --train-dir results/silicon-tutorial/si --batch-size 1`
The following error pops up
`FileNotFoundError: [Errno 2] No such file or directory: 'results/silicon-tutorial/si/best_model.pth'`
And the subsequent code after that does not run.

There is no best_model.pth also after running the training command
![image](https://github.com/user-attachments/assets/860e3b14-4824-4e1c-9673-432205f8ab02)

I might be missing something. Please have a look.

Thank you for your time and consideration,
Piyush

**Reply by catastropiyush** [reply]
Hello, @Linux-cpp-lisp can you please look into this?

Thank you for your time,
Piyush

**Reply by Linux-cpp-lisp** [reply]
If there is no best_model.pth, your training likely did not actually run a full epoch.  Did you see it get through a validation after epoch in the logs?

**Comment by catastropiyush** [comment]
After running this 
`!rm -rf ./results
!nequip-train allegro/configs/tutorial.yaml --equivariance-test`

![image](https://github.com/user-attachments/assets/04c4c265-455f-477f-9a1a-5337a93ce045)
After that 
![image](https://github.com/user-attachments/assets/89499618-a5d9-4e0b-b358-2df4e56caa64)

I think the error is in the first train command.
Can you please look into it. @Linux-cpp-lisp 
Thank you for your time,
Piyush
